app/jobs/report_job.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# Renders a report and saves it to a temporary file.
+class ReportJob < ActiveJob::Base
```

Comment:
Yes, sorry for the detour. The first instance of a job can be used for direct execution for for storing with `perform_later` which means that Sidekiq will pick it up at some point. And I guess that Sidekiq will instantiate a similar object, not necessarily an ActiveJob but something Sidekiq specific.

Anyway, we are not using Sidekiq in the end. But I still found it a useful interface and abstraction. It's still a job that's run somewhere else. We just implemented our own forked job runner. Maybe it would have been cleaner to implement that as an ActiveJob middleware but ... also more to learn.